src/world/mod.rs

```diff
@@ -387,8 +387,10 @@ impl World {
         self.write_resource()
     }
 
-    /// Allows building an entity with its
-    /// components.
+    /// Allows building an entity with its components.
+    ///
+    /// **You have to ensure that the component storages you're
```

Comment:
or what? I'd expect this to be something enforced by the type system

Reply:
Or it would panic, which is very unfortunate. The problem is that you can't borrow e.g. a resource otherwise. So we have the choice between taking `&mut self` and not allowing resource access (which is a common use case) or taking `&self` and risking that someone has borrowed that storage.

Reply:
The core problem of all this is that the borrow checker cannot track our resource entries :( (which makes sense, but still)

Reply:
I'll solve this before I release a new version by renaming this method to `create_entity_unchecked` and just having a `create_entity` accept a mutable world.